uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/PortletPrefsRESTController.java

```diff
@@ -0,0 +1,676 @@
+import org.apereo.portal.portlet.registry.IPortletEntityRegistry;
+import org.apereo.portal.security.AuthorizationPrincipalHelper;
+import org.apereo.portal.security.IAuthorizationPrincipal;
+import org.apereo.portal.security.IPerson;
+import org.apereo.portal.security.IPersonManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+/**
+ * PortletPrefsRESTController provides REST targets for preference operations for soffits and
+ * webcomponents.
+ */
+@Controller
+public class PortletPrefsRESTController {
+
+    protected final Log logger = LogFactory.getLog(getClass());
+    private ObjectMapper mapper;
+
+    private IPersonManager personManager;
+
+    private final boolean DEFINITION_MODE = true;
+    private final boolean ENTITY_MODE = false;
+
+    @Autowired
+    public void setPersonManager(IPersonManager personManager) {
+        this.personManager = personManager;
+    }
+
+    private IPortletDefinitionDao portletDao;
+
+    @Autowired
+    public void setPortletDao(IPortletDefinitionDao portletDao) {
+        this.portletDao = portletDao;
+    }
+
+    private IPortletEntityRegistry portletEntityRegistry;
+
+    @Autowired
+    public void setPortletEntityRegistry(IPortletEntityRegistry portletEntityRegistry) {
+        this.portletEntityRegistry = portletEntityRegistry;
+    }
+
+    private PortletPreferencesFactory portletPreferencesFactory;
+
+    @Autowired
+    @Qualifier("portletPreferencesFactory")
+    public void setPortletPreferencesFactoryAPI(
+            PortletPreferencesFactory portletPreferencesFactory) {
+        this.portletPreferencesFactory = portletPreferencesFactory;
+    }
+
+    public PortletPrefsRESTController() {
+        this.mapper = new ObjectMapper();
+    }
+
+    /**
+     * Get uPortal Entity information for portlet/soffit/webcomponent. This method provides a REST
+     * interface for obtaining uPortal portlet/soffit/webcomponent metadata
+     *
+     * <p>The path for this method is /prefs/getentity/fname. The fname is a string representing the
+     * fname of the portlet/soffit/webcomponent to find the entity information for
+     */
+    /*non-javadoc
+     * I used this primarily for testing, could be useful given that it returns the windowstate, etc
+     * up to you if you want to keep it or modify it
+     * */
+    @RequestMapping(value = "/prefs/getentity/{fname}", method = RequestMethod.GET)
```

Comment:
We shouldn't use verbs in URIs.

I'm not sure this method is a good fit for this `@Controller` class.

Reply:
@drewwills I wasn't sure that the method was appropriate either, which is why I put the comment above it. Do you want me to remove it?

Reply:
@drewwills note that if you want them to know their own windowId, we will need some sort of replacement call to allow them access to that.